workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -200,7 +200,9 @@ def __init__(
         self._prephysics_only_diagnostic_ml: bool = getattr(
             getattr(config, "prephysics"), "diagnostic_ml", False
         )
-        self._postphysics_only_diagnostic_ml: bool = config.scikit_learn.diagnostic_ml
+        self._postphysics_only_diagnostic_ml: bool = getattr(
```

Comment:
This is awkward. This is the downside of making MachineLearningConfig an optional. It essentially moves the conditional logic from `prepare_config.py` to here, although I think the simplifications to prepare_config are worth it in this case. Ideally this logic could be moved to where the `Steppers` are constructed https://github.com/VulcanClimateModeling/fv3net/blob/4fc74a307e6c88e12c738df9d8484b8f53499e0d/workflows/prognostic_c48_run/runtime/loop.py#L234-L234

Reply:
Not a required change, but something which could be addressed more elegantly. e.g. the MLStepper could be passed `do_diagnostic_ml` and change whether the ML outputs are put into "diagnostics" or "tendencies".

Reply:
Okay, definitely agree it would be cleaner to have the `do_diagnostic_ml` passed to the relevant Stepper, but this is a non-trivial change and I'm not sure our tests would catch a bug introduced by the change so I'm going to do it in a separate PR.